### ISSUE TITLE
OneSiganl.getTags - TypeError: Cannot read property 'tags' of undefined

### ISSUE THREAD

**Issue by steveaz** [OneSignal/OneSignal-Website-SDK#477]
**Description:**

I have something strange happening for the OneSignal.getTags function.

It appears to work fine with no issues, until I unsubscribe and attempt to resubscribe to receive push notifications.

The error I receive is: OneSignal.js:414 Uncaught (in promise) TypeError: Cannot read property 'tags' of undefined

and the code is highlighting this area in OneSignal.js: const { tags } = await OneSignalApi.getPlayer(appId, deviceId);

**Steps to Reproduce Issue:**

1. Subscribe to receive push notifications.
2. SendTags to this session
2. Click the "Lock" icon on Chrome and remove the push notifications permissions.
3. Reload the page

**Anything else:**

When I go into Chrome's Application viewer and Clear Storage, the getTags function returns to working as expected.

This is the function I am trying to run:

OneSignal.push(function() {
		OneSignal.getTags().then(function(tags) {
				console.log(tags);
			});
});

**Comment by rgomezp** [comment]
Hello,
I was unfortunately not able to reproduce the issue. This could be an issue in your initialization or perhaps a bug on our end. Try following the steps to [debug](https://documentation.onesignal.com/docs/troubleshooting-web-push#section-debugging-using-browser-developer-tools) using the browser developer tools. Hopefully this will provide additional insight. Also, if you send your entire code, I may be able to successfully reproduce the issue.

**Comment by rgomezp** [comment]
Closing due to no response.

**Comment by steveaz** [comment]
Hi Rgomezp,

Thank you for the post.

I have tried to enable the debug using the browser developer tools and the only thing I see that stands out is this line:

(Host) » notifyButtonStateChange: {from: "uninitialized", to: "unsubscribed"}

This should be saying Subscribed.

For the gettags error i'm seeing, it seems to be when subscribing and unsubscribing without clearing the entire cache and closing and re-opening the browser. Is there nothing else that can be done to fix this? The next time I receive this error I will update this post with what I see.

Since debugging this over and over and over, I now have a completely different error. I seem to no longer be able to subscribe to Push Notifications.

I receive a 400 error stating "{"errors":["No user with this id found"]}"

https://onesignal.com/api/v1/players/426757d2-c778-47a6-ae9d-9729cfc6a4eb

Here is the data my browser is trying to send:

app_id: "9c9c6576-957b-43c4-b00d-13693d4daf65"
browser_name: "Chrome"
browser_version: 72
delivery_platform: 5
device_model: "Win32"
device_os: 72
device_platform: "desktop"
device_type: 5
identifier: "https://fcm.googleapis.com/fcm/send/eW7DYv4AiXY:APA91bG6_b6a5BvZks4D8zaIlIDRT9Da08mR9rb_feww76ikielDG9ubhNCB95bkabMBC37glzvzCXeE4Kau_-KWFdLCcaSAEpE7oyAQVsgOrxvDSUglsql2BMqLW5m3W-AwA7jaXFZY"
language: "en"
notification_types: 1
operating_system: "Microsoft Windows"
operating_system_version: "10"
sdk: "150506"
timezone: -25200
web_auth: "8v4w1FGon/+Aa7o6U80pRA=="
web_p256: "BO6emhDKLGDtARBTaYSFiAuhZvv9sOerTnR1qRxjaoCycWQBVMcWFowiSGzMPt1WVZnU5TVZt/Jj9YEpV0JejQ4="

This seems to be failing for the most basic of calls and I'm wondering if maybe hte OneSignal system has blocked me or what could be happening here?

I've created a simple test page with a simple implimentation of OneSignal here: https://vipairsoft.net/test.php

Any insight would be greatly appreciated!!!
